server/pulp/server/controllers/importer.py

```diff
@@ -35,16 +37,28 @@ def build_resource_tag(repo_id, importer_type_id):
 
 def clean_config_dict(config):
     """
-    Remove keys from a dict that have a value None.
+    Ensure that the config is ready to be passed to the plugin. There are two things that must
+    occur:
+        1. Remove keys from the dict that have a value None.
+        2. If they exist, move basic auth credentials from the feed_url to the config.
 
     :param config: configuration for plugin
     :type  config: dict
 
     :return: config without the keys whose values were None
     :rtype:  dict:
     """
-
     if config is not None:
+        feed_url = config.get('feed')
+        if feed_url:
+            parsed = urlparse.urlparse(feed_url)
+            if parsed.username and parsed.password:
+                at = parsed.netloc.find('@')
+                new_url = parsed._replace(netloc=parsed.netloc[at + 1:])
```

Comment:
This implementation is broken, and the rfind approach (or something like it) needs to be used, since username and password can contain `@`. hostnames, cannot, so the rightmost `@` must be used to partition the hostname off from the user/pass components. Also, it's possible for only the username to be set (where netloc is `username@hostname`) and for only the password to be set (where netloc is `:password@hostname`), so the conditional on line 55 should be `if parsed.username or parsed.password`.

```python
In [1]: from urllib import parse

In [2]: parsed = parse.urlparse('scheme://user@name:pass@word@hostname/path')

In [3]: at = parsed.netloc.find('@')

In [4]: parsed._replace(netloc=parsed.netloc[at + 1:])
Out[4]: ParseResult(scheme='scheme', netloc='name:pass@word@hostname', path='/path', params='', query='', fragment='')

In [5]: parsed._replace(netloc=parsed.netloc.rsplit('@', 1)[1])
Out[5]: ParseResult(scheme='scheme', netloc='hostname', path='/path', params='', query='', fragment='')

In [6]: parsed.netloc, parsed.username, parsed.password
Out[6]: ('user@name:pass@word@hostname', 'user@name', 'pass@word')

```
Note that with the `at = ` approach, the netloc in the new url is wrong.